File: Elmah.Contrib.EntityFramework/ElmahContext.fs
namespace Elmah.Contrib.EntityFramework

open System.Data.Entity

type public ElmahContext (nameOrConnectionString : string) =
    inherit DbContext (nameOrConnectionString)
   
    override this.OnModelCreating(modelBuilder : DbModelBuilder) =
        modelBuilder.Entity<ElmahError>().ToTable("ELMAH_Error") |> ignore
        
        base.OnModelCreating(modelBuilder)

    [<DefaultValue>]
    val mutable _errors : DbSet<ElmahError>
    
    member public this.Errors
        with get() = this._errors 
        and set(v) = this._errors <- v

File: Elmah.Contrib.EntityFramework/ElmahContext.cs
using System.Data.Entity;

namespace Elmah.Contrib.EntityFramework
{
	public class ElmahContext : DbContext
	{
		private readonly string _tableName;

		static ElmahContext()
		{
			Database.SetInitializer(new NullDatabaseInitializer<ElmahContext>());
		}

		public ElmahContext(string nameOrConnectionString)
			: this(nameOrConnectionString, "ELMAH_Error")
		{
		}

		public ElmahContext(string nameOrConnectionString, string tableName)
			: base(nameOrConnectionString)
		{
			_tableName = tableName;
		}

		protected override void OnModelCreating(DbModelBuilder modelBuilder)
		{
			modelBuilder.Configurations.Add(GetConfiguration());
		}

		protected virtual EntityTypeConfiguration<ElmahError> GetConfiguration()
		{
			return new ElmahErrorConfiguration(_tableName);
		}

		public virtual DbSet<ElmahError> ElmahErrors { get; set; }
	}
}